osu.Game/Beatmaps/BeatmapDifficultyCache.cs

```diff
@@ -266,8 +267,11 @@ private void updateBindable([NotNull] BindableStarDifficulty bindable, [CanBeNul
                     // We're on a threadpool thread, but we should exit back to the update thread so consumers can safely handle value-changed events.
                     Schedule(() =>
                     {
-                        if (!cancellationToken.IsCancellationRequested && t.Result != null)
-                            bindable.Value = t.Result;
+                        if (cancellationToken.IsCancellationRequested)
+                            return;
+
+                        if (t.WaitSafelyForResult() is StarDifficulty sd)
```

Comment:
Not too sure of using `is` as a null check... Would probably prefer the long-winded but obvious variant.

Same goes for a few other places in this pull (`ChannelManager`, etc.)

Reply:
hmm, we already do this in some other places and i've never found it too offensive.

```csharp
var sd = t.WaitSafelyForResult();

if (sd != null)
{

}
```

Is this what you'd prefer?

Reply:
I think explicit null check as above is clunkier but better, but I won't insist on it. Dunno, just feels like null checking is a semi-hidden feature of the pattern matching syntax as is.

Reply:
Is fair. Should probably reach a consensus on this and apply codebase wise, since I know this is in a few other places.